src/Catalog/Extensions/IDataRecordExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace System.Data
+{
+    /// <summary>
+    /// Extension methods that make working with <see cref="IDataRecord"/> more convenient.
+    /// </summary>
+    public static class IDataRecordExtensions
+    {
+        public static DateTime ReadNullableUtcDateTime(this IDataRecord dataRecord, string columnName)
+        {
+            if (dataRecord == null)
+            {
+                throw new ArgumentNullException(nameof(dataRecord));
+            }
+
+            if (columnName == null)
+            {
+                throw new ArgumentNullException(nameof(columnName));
+            }
+
+            return (dataRecord[columnName] == DBNull.Value ? DateTime.MinValue : ReadDateTime(dataRecord, columnName)).ForceUtc();
+        }
+
+        public static DateTime ReadDateTime(this IDataRecord dataRecord, string columnName)
+        {
+            if (dataRecord == null)
+            {
+                throw new ArgumentNullException(nameof(dataRecord));
+            }
+
+            if (columnName == null)
+            {
+                throw new ArgumentNullException(nameof(columnName));
+            }
+
+            return dataRecord.GetDateTime(dataRecord.GetOrdinal(columnName));
```

Comment:
What is the behavior of the column type if not DateTime?

Reply:
That would depend on the concrete implementation of `IDataRecord.GetDateTime`.